<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Connectome Basic Workflow • Connectome</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Connectome Basic Workflow">
<meta property="og:description" content="Connectome">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">Connectome</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.2.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/01%20Connectome%20Workflow.html">Connectome Basic Workflow</a>
    </li>
    <li>
      <a href="../articles/02%20Differential%20Connectomics.html">Differential Connectomics</a>
    </li>
    <li>
      <a href="../articles/03%20Centrality%20Plots.html">Centrality Plots</a>
    </li>
    <li>
      <a href="../articles/04%20Comparative%20and%20Longitudinal%20Connectomics.html">Comparative and Longitudinal Connectomics</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Connectome Basic Workflow</h1>
                        <h4 class="author">Micha Sam Brickman Raredon</h4>
            
            <h4 class="date">2020-05-13</h4>
      
      
      <div class="hidden name"><code>01 Connectome Workflow.Rmd</code></div>

    </div>

    
    
<p>Connectome is an R package to automate ligand-receptor mapping and visualization in single-cell data. It is built to work with Seurat from Satija Lab.</p>
<div id="load-dependencies" class="section level2">
<h2 class="hasAnchor">
<a href="#load-dependencies" class="anchor"></a>Load dependencies</h2>
<div class="sourceCode" id="cb1"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">Seurat</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">SeuratData</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">Connectome</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">ggplot2</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">cowplot</span>)</pre></body></html></div>
</div>
<div id="data-preparation" class="section level2">
<h2 class="hasAnchor">
<a href="#data-preparation" class="anchor"></a>Data Preparation</h2>
<p><code>CreateConnectome</code> uses the active identity slot in a Seurat 3.0 object to define “nodes” for network creation and network analysis.</p>
<p>Node parcellation should be done carefully as it will have a non-trivial effect on downstream connectomics.</p>
</div>
<div id="load-data" class="section level2">
<h2 class="hasAnchor">
<a href="#load-data" class="anchor"></a>Load Data</h2>
<p>Here, we run <em>Connectome</em> on the cross-platform Pancreas data distributed by SeuratData:</p>
<div class="sourceCode" id="cb2"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/pkg/SeuratData/man/InstallData.html">InstallData</a></span>(<span class="st">'panc8'</span>)
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="st">'panc8'</span>)
<span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span>(<span class="fu"><a href="https://rdrr.io/pkg/Seurat/man/Idents.html">Idents</a></span>(<span class="no">panc8</span>))
<span class="co">#&gt; </span>
<span class="co">#&gt;     celseq    celseq2 fluidigmc1     indrop  smartseq2 </span>
<span class="co">#&gt;       1004       2285        638       8569       2394</span>
<span class="fu"><a href="https://rdrr.io/pkg/Seurat/man/Idents.html">Idents</a></span>(<span class="no">panc8</span>) <span class="kw">&lt;-</span> <span class="no">panc8</span><span class="kw">[[</span><span class="st">'celltype'</span>]]
<span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span>(<span class="fu"><a href="https://rdrr.io/pkg/Seurat/man/Idents.html">Idents</a></span>(<span class="no">panc8</span>))
<span class="co">#&gt; </span>
<span class="co">#&gt;              gamma             acinar              alpha </span>
<span class="co">#&gt;                625               1864               4615 </span>
<span class="co">#&gt;              delta               beta             ductal </span>
<span class="co">#&gt;               1013               3679               1954 </span>
<span class="co">#&gt;        endothelial activated_stellate            schwann </span>
<span class="co">#&gt;                296                474                 25 </span>
<span class="co">#&gt;               mast         macrophage            epsilon </span>
<span class="co">#&gt;                 56                 79                 30 </span>
<span class="co">#&gt; quiescent_stellate </span>
<span class="co">#&gt;                180</span></pre></body></html></div>
</div>
<div id="scale-and-make-connectome" class="section level2">
<h2 class="hasAnchor">
<a href="#scale-and-make-connectome" class="anchor"></a>Scale and make connectome</h2>
<p>Here, we normalize the data, identify the ligand and receptor genes that have mapped in the object, scale those genes, and generate the connectome. To accelerate computation time, we omit calculation of Wilcoxon rank p-values and gene-wsie diagnostic odds ratios. We also limit our connectomic analysis to only those clusters with &gt; 75 cells captured, since small clusters may not accurately regress to reliable mean values.</p>
<div class="sourceCode" id="cb3"><html><body><pre class="r"><span class="no">panc8</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Seurat/man/NormalizeData.html">NormalizeData</a></span>(<span class="no">panc8</span>)
<span class="no">connectome.genes</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sets.html">union</a></span>(<span class="kw pkg">Connectome</span><span class="kw ns">::</span><span class="no">ncomms8866_human</span>$<span class="no">Ligand.ApprovedSymbol</span>,<span class="kw pkg">Connectome</span><span class="kw ns">::</span><span class="no">ncomms8866_human</span>$<span class="no">Receptor.ApprovedSymbol</span>)
<span class="no">genes</span> <span class="kw">&lt;-</span> <span class="no">connectome.genes</span>[<span class="no">connectome.genes</span> <span class="kw">%in%</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="no">panc8</span>)]
<span class="no">panc8</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Seurat/man/ScaleData.html">ScaleData</a></span>(<span class="no">panc8</span>,<span class="kw">features</span> <span class="kw">=</span> <span class="no">genes</span>)
<span class="no">panc8.con</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/CreateConnectome.html">CreateConnectome</a></span>(<span class="no">panc8</span>,<span class="kw">species</span> <span class="kw">=</span> <span class="st">'human'</span>,<span class="kw">min.cells.per.ident</span> <span class="kw">=</span> <span class="fl">75</span>,<span class="kw">p.values</span> <span class="kw">=</span> <span class="no">F</span>,<span class="kw">calculate.DOR</span> <span class="kw">=</span> <span class="no">F</span>)</pre></body></html></div>
</div>
<div id="filter-to-edges-of-interest" class="section level2">
<h2 class="hasAnchor">
<a href="#filter-to-edges-of-interest" class="anchor"></a>Filter to edges of interest</h2>
<p>This is a critical step which is both dataset- and scientific question-dependent. First, let’s look at the distribution of ligand and receptor z-scores and percentage expression:</p>
<div class="sourceCode" id="cb4"><html><body><pre class="r"><span class="co"># Change density plot line colors by groups</span>
<span class="no">p1</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span>(<span class="no">panc8.con</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(<span class="kw">x</span><span class="kw">=</span><span class="no">ligand.scale</span>)) + <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_density.html">geom_density</a></span>() + <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ggtitle</a></span>(<span class="st">'Ligand.scale'</span>)
<span class="no">p2</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span>(<span class="no">panc8.con</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(<span class="kw">x</span><span class="kw">=</span><span class="no">recept.scale</span>)) + <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_density.html">geom_density</a></span>() + <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ggtitle</a></span>(<span class="st">'Recept.scale'</span>)
<span class="no">p3</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span>(<span class="no">panc8.con</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(<span class="kw">x</span><span class="kw">=</span><span class="no">percent.target</span>)) + <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_density.html">geom_density</a></span>() + <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ggtitle</a></span>(<span class="st">'Percent.target'</span>)
<span class="no">p4</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span>(<span class="no">panc8.con</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(<span class="kw">x</span><span class="kw">=</span><span class="no">percent.source</span>)) + <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_density.html">geom_density</a></span>() + <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ggtitle</a></span>(<span class="st">'Percent.source'</span>)
<span class="fu"><a href="https://rdrr.io/pkg/cowplot/man/plot_grid.html">plot_grid</a></span>(<span class="no">p1</span>,<span class="no">p2</span>,<span class="no">p3</span>,<span class="no">p4</span>)</pre></body></html></div>
<p><img src="01%20Connectome%20Workflow_files/figure-html/unnamed-chunk-4-1.png" width="960" style="display: block; margin: auto;"></p>
<p>Given these outputs, let’s filter the data to only include edges with a ligand and receptor z-score above 0.25 (thereby emphasizes cell-type specific communication patterns), and with both the ligand and receptor expressed in at least 10% of the cells in their respective cluster (thereby increasing statistical confidence):</p>
<div class="sourceCode" id="cb5"><html><body><pre class="r"><span class="no">panc8.con2</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/FilterConnectome.html">FilterConnectome</a></span>(<span class="no">panc8.con</span>,<span class="kw">min.pct</span> <span class="kw">=</span> <span class="fl">0.1</span>,<span class="kw">min.z</span> <span class="kw">=</span> <span class="fl">0.25</span>,<span class="kw">remove.na</span> <span class="kw">=</span> <span class="no">T</span>)</pre></body></html></div>
<p>We are now ready to begin making plots of this single-tissue system.</p>
</div>
<div id="network-plot" class="section level2">
<h2 class="hasAnchor">
<a href="#network-plot" class="anchor"></a>Network plot</h2>
<p>This is the simplest form of network visualization. This function will take as input any parameters which can be passed to <code>FilterConnectome</code>, thereby allowing in-line filtration. Here, we exclusievely look at cell-cell edges involving VEGFA. We also show the effect of filtration based on percent expression:</p>
<div class="sourceCode" id="cb6"><html><body><pre class="r"><span class="no">p1</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/NetworkPlot.html">NetworkPlot</a></span>(<span class="no">panc8.con2</span>,<span class="kw">features</span> <span class="kw">=</span> <span class="st">'VEGFA'</span>,<span class="kw">min.pct</span> <span class="kw">=</span> <span class="fl">0.1</span>,<span class="kw">weight.attribute</span> <span class="kw">=</span> <span class="st">'weight_sc'</span>,<span class="kw">include.all.nodes</span> <span class="kw">=</span> <span class="no">T</span>)
<span class="no">p2</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/NetworkPlot.html">NetworkPlot</a></span>(<span class="no">panc8.con2</span>,<span class="kw">features</span> <span class="kw">=</span> <span class="st">'VEGFA'</span>,<span class="kw">min.pct</span> <span class="kw">=</span> <span class="fl">0.75</span>,<span class="kw">weight.attribute</span> <span class="kw">=</span> <span class="st">'weight_sc'</span>,<span class="kw">include.all.nodes</span> <span class="kw">=</span> <span class="no">T</span>)</pre></body></html></div>
<div class="sourceCode" id="cb7"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/pkg/cowplot/man/plot_grid.html">plot_grid</a></span>(<span class="no">p1</span>,<span class="no">p2</span>,<span class="kw">nrow</span><span class="kw">=</span><span class="fl">1</span>)</pre></body></html></div>
<p><img src="01%20Connectome%20Workflow_files/figure-html/unnamed-chunk-7-1.png" width="960" style="display: block; margin: auto;"></p>
</div>
<div id="centrality-analysis" class="section level2">
<h2 class="hasAnchor">
<a href="#centrality-analysis" class="anchor"></a>Centrality analysis</h2>
<p>Built into <code>Connectome</code> is a function called <code>Centrality</code>, which performs a graph-theory based analysis of the input network. The user specifies how to subdivide the input network using the <code>group.by</code> argument. For each sub-network, cell types are ranked by their cumulative incoming/outgoing edgeweight and their Kleinberg directional centrality. This allows quick identification of top ligand producers and correlating receptor receivers, in each signaling subnetwork. Here, we group cell-cell communication in the pancreas by signaling family (‘mode’):</p>
<div class="sourceCode" id="cb8"><html><body><pre class="r"><span class="fu"><a href="../reference/Centrality.html">Centrality</a></span>(<span class="no">panc8.con2</span>,
           <span class="kw">modes.include</span> <span class="kw">=</span> <span class="kw">NULL</span>,
           <span class="kw">min.z</span> <span class="kw">=</span> <span class="kw">NULL</span>,
           <span class="kw">weight.attribute</span> <span class="kw">=</span> <span class="st">'weight_sc'</span>,
           <span class="kw">group.by</span> <span class="kw">=</span> <span class="st">'mode'</span>)</pre></body></html></div>
<p><img src="01%20Connectome%20Workflow_files/figure-html/unnamed-chunk-8-1.png" width="1920"></p>
<p><code>Centrality</code> can also be used to look more closely at specific signaling families of particular interest, grouped here by mechanism:</p>
<div class="sourceCode" id="cb9"><html><body><pre class="r"><span class="fu"><a href="../reference/Centrality.html">Centrality</a></span>(<span class="no">panc8.con2</span>,
           <span class="kw">modes.include</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">'VEGF'</span>),
           <span class="kw">weight.attribute</span> <span class="kw">=</span> <span class="st">'weight_sc'</span>,
           <span class="kw">min.z</span> <span class="kw">=</span> <span class="fl">0</span>,
           <span class="kw">group.by</span> <span class="kw">=</span> <span class="st">'mechanism'</span>)</pre></body></html></div>
<p><img src="01%20Connectome%20Workflow_files/figure-html/unnamed-chunk-9-1.png" width="960"></p>
</div>
<div id="cellcellscatter" class="section level2">
<h2 class="hasAnchor">
<a href="#cellcellscatter" class="anchor"></a>CellCellScatter</h2>
<p>If a particular cell-to-cell interaction is of particular interest, <code>CellCellScatter</code> can be useful to immediately identify top signaling modes between those two cells. The x-axis is the expression of the receptor on the receiving cell type, while the y-axis is the expression of the ligand on the sending cell type:</p>
<div class="sourceCode" id="cb10"><html><body><pre class="r"><span class="no">p1</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/CellCellScatter.html">CellCellScatter</a></span>(<span class="no">panc8.con2</span>,<span class="kw">sources.include</span> <span class="kw">=</span> <span class="st">'endothelial'</span>,<span class="kw">targets.include</span> <span class="kw">=</span> <span class="st">'activated_stellate'</span>,
                <span class="kw">label.threshold</span> <span class="kw">=</span> <span class="fl">3</span>,
                <span class="kw">weight.attribute</span> <span class="kw">=</span> <span class="st">'weight_sc'</span>,<span class="kw">min.pct</span> <span class="kw">=</span> <span class="fl">0.25</span>,<span class="kw">min.z</span> <span class="kw">=</span> <span class="fl">2</span>)
<span class="no">p1</span> <span class="kw">&lt;-</span> <span class="no">p1</span> + <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/lims.html">xlim</a></span>(<span class="fl">0</span>,<span class="fl">NA</span>) + <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/lims.html">ylim</a></span>(<span class="fl">0</span>,<span class="fl">NA</span>)
<span class="no">p1</span></pre></body></html></div>
<p><img src="01%20Connectome%20Workflow_files/figure-html/unnamed-chunk-10-1.png" width="960" style="display: block; margin: auto;"> From the above plot, we can see that endothelial cells are prominently expressing DLL4, which might be sensed by the NOTCH3 receptor on activated stellate cells.</p>
</div>
<div id="signalscatter" class="section level2">
<h2 class="hasAnchor">
<a href="#signalscatter" class="anchor"></a>SignalScatter</h2>
<p>It might then be important to ask if the DLL4-NOTCH3 interaction from the above example is exclusive to the endothelial-activated stellate cell vector. It is possible that although it is a top mechanism between these two cells, there may be another cell-cell vector which more strongly utilizes this pathway. To test this hypothesis, we can use ‘SignalScatter’ to investigate the entire DLL4-NOTCH3 network across all cell types. We also choose to include additional ligands which can be sensed by NOTCH3:</p>
<div class="sourceCode" id="cb11"><html><body><pre class="r"><span class="no">p2</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/SignalScatter.html">SignalScatter</a></span>(<span class="no">panc8.con2</span>, <span class="kw">features</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">'JAG1'</span>,<span class="st">'JAG2'</span>,<span class="st">'DLL4'</span>),<span class="kw">label.threshold</span> <span class="kw">=</span> <span class="fl">1</span>,<span class="kw">weight.attribute</span> <span class="kw">=</span> <span class="st">'weight_sc'</span>,<span class="kw">min.z</span> <span class="kw">=</span> <span class="fl">2</span>)
<span class="no">p2</span> <span class="kw">&lt;-</span> <span class="no">p2</span> + <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/lims.html">xlim</a></span>(<span class="fl">2</span>,<span class="fl">NA</span>) + <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/lims.html">ylim</a></span>(<span class="fl">2</span>,<span class="fl">NA</span>)
<span class="no">p2</span></pre></body></html></div>
<p><img src="01%20Connectome%20Workflow_files/figure-html/unnamed-chunk-11-1.png" width="960" style="display: block; margin: auto;"> From this plot, we see that DLL4-NOTCH3 is most likely mediating communication between endothelial cells and <strong>both</strong> activated and quiescent stellate cells. Further, we can see that NOTCH4 is likley mediating endothelial autocrine communication, both via DLL4 and other ligands (JAG1 and JAG2).</p>
</div>
<div id="exploring-connectomic-data-using-circosplot" class="section level2">
<h2 class="hasAnchor">
<a href="#exploring-connectomic-data-using-circosplot" class="anchor"></a>Exploring connectomic data using CircosPlot</h2>
<p>The clearest way we have found to visualize cell-cell connectivity in scRNAseq data is through circos plots made with the R package <code>circlize</code>. We have therefore built a versatile <code>CircosPlot</code> function into the <code>Connectome</code> package.</p>
<p>First, let’s select the top 5 signaling vectors for each cell-cell vector:</p>
<div class="sourceCode" id="cb12"><html><body><pre class="r"><span class="no">test</span> <span class="kw">&lt;-</span> <span class="no">panc8.con2</span>
<span class="no">test</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span>(<span class="no">test</span> <span class="kw">%&gt;%</span> <span class="fu">group_by</span>(<span class="no">vector</span>) <span class="kw">%&gt;%</span> <span class="fu">top_n</span>(<span class="fl">5</span>,<span class="no">weight_sc</span>))</pre></body></html></div>
<p>Then, let’s focus in on specific cell types of interest:</p>
<div class="sourceCode" id="cb13"><html><body><pre class="r"><span class="no">cells.of.interest</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">'endothelial'</span>,<span class="st">'activated_stellate'</span>,<span class="st">'quiescent_stellate'</span>,<span class="st">'macrophage'</span>)</pre></body></html></div>
<p>CircosPlot can make 4 distinct plot types, with subtle quantitative differences:</p>
<div class="sourceCode" id="cb14"><html><body><pre class="r"><span class="co"># Using edgeweights from normalized slot:</span>
<span class="fu"><a href="../reference/CircosPlot.html">CircosPlot</a></span>(<span class="no">test</span>,<span class="kw">weight.attribute</span> <span class="kw">=</span> <span class="st">'weight_norm'</span>,<span class="kw">sources.include</span> <span class="kw">=</span> <span class="no">cells.of.interest</span>,<span class="kw">targets.include</span> <span class="kw">=</span> <span class="no">cells.of.interest</span>,<span class="kw">lab.cex</span> <span class="kw">=</span> <span class="fl">0.6</span>,<span class="kw">title</span> <span class="kw">=</span> <span class="st">'Edgeweights from normalized slot'</span>)
<span class="co"># Using edgeweights from scaled slot:</span>
<span class="fu"><a href="../reference/CircosPlot.html">CircosPlot</a></span>(<span class="no">test</span>,<span class="kw">weight.attribute</span> <span class="kw">=</span> <span class="st">'weight_sc'</span>,<span class="kw">sources.include</span> <span class="kw">=</span> <span class="no">cells.of.interest</span>,<span class="kw">targets.include</span> <span class="kw">=</span> <span class="no">cells.of.interest</span>,<span class="kw">lab.cex</span> <span class="kw">=</span> <span class="fl">0.6</span>,<span class="kw">title</span> <span class="kw">=</span> <span class="st">'Edgeweights from scaled slot'</span>)
<span class="co"># Using separate ligand and receptor expression (from normalized slot)</span>
<span class="fu"><a href="../reference/CircosPlot.html">CircosPlot</a></span>(<span class="no">test</span>,<span class="kw">weight.attribute</span> <span class="kw">=</span> <span class="st">'weight_norm'</span>,<span class="kw">sources.include</span> <span class="kw">=</span> <span class="no">cells.of.interest</span>,<span class="kw">targets.include</span> <span class="kw">=</span> <span class="no">cells.of.interest</span>,<span class="kw">balanced.edges</span> <span class="kw">=</span> <span class="no">F</span>,<span class="kw">lab.cex</span> <span class="kw">=</span> <span class="fl">0.6</span>,<span class="kw">title</span> <span class="kw">=</span> <span class="st">'Ligand vs. receptor expression (from normalized slot)'</span>)
<span class="co"># Using separate ligand and receptor expression (from scaled slot)</span>
<span class="fu"><a href="../reference/CircosPlot.html">CircosPlot</a></span>(<span class="no">test</span>,<span class="kw">weight.attribute</span> <span class="kw">=</span> <span class="st">'weight_sc'</span>,<span class="kw">sources.include</span> <span class="kw">=</span> <span class="no">cells.of.interest</span>,<span class="kw">targets.include</span> <span class="kw">=</span> <span class="no">cells.of.interest</span>,<span class="kw">balanced.edges</span> <span class="kw">=</span> <span class="no">F</span>,<span class="kw">lab.cex</span> <span class="kw">=</span> <span class="fl">0.6</span>,<span class="kw">title</span> <span class="kw">=</span> <span class="st">'Ligand vs. receptor expression (from scaled slot)'</span>)</pre></body></html></div>
<p><img src="01%20Connectome%20Workflow_files/figure-html/unnamed-chunk-14-1.png" width="960" style="display: block; margin: auto;"><img src="01%20Connectome%20Workflow_files/figure-html/unnamed-chunk-14-2.png" width="960" style="display: block; margin: auto;"><img src="01%20Connectome%20Workflow_files/figure-html/unnamed-chunk-14-3.png" width="960" style="display: block; margin: auto;"><img src="01%20Connectome%20Workflow_files/figure-html/unnamed-chunk-14-4.png" width="960" style="display: block; margin: auto;"></p>
<p>Although the 4 plots look similar, each is leveraging slightly different quantitative information. Appropriate use should be determined by the scientific question of interest.</p>
</div>
<div id="niche-wise-investigation" class="section level2">
<h2 class="hasAnchor">
<a href="#niche-wise-investigation" class="anchor"></a>Niche-wise investigation:</h2>
<p><code>Connectome</code> can be used to visualize the cell-type specific “niche” network for a given cell type, showing all possible communication pathways converging on a given cell. This can be done leveraging the <code>targets.include</code> argument built into many of the above functions. Here, we choose to look at the endothelial niche in the pancreas:</p>
<div class="sourceCode" id="cb15"><html><body><pre class="r"><span class="fu"><a href="../reference/CircosPlot.html">CircosPlot</a></span>(<span class="no">test</span>,<span class="kw">targets.include</span> <span class="kw">=</span> <span class="st">'endothelial'</span>,<span class="kw">lab.cex</span> <span class="kw">=</span> <span class="fl">0.6</span>)</pre></body></html></div>
<p><img src="01%20Connectome%20Workflow_files/figure-html/unnamed-chunk-15-1.png" width="960" style="display: block; margin: auto;"></p>
</div>
<div id="source-wise-investigation" class="section level2">
<h2 class="hasAnchor">
<a href="#source-wise-investigation" class="anchor"></a>Source-wise investigation:</h2>
<p>Similarly, the argument <code>sources.include</code> can be used to visualize all edges coming <em>from</em> a given cell type. This plot shows all of the cell-cell edges <em>originating</em> from endothelial cells in the pancreas:</p>
<div class="sourceCode" id="cb16"><html><body><pre class="r"><span class="fu"><a href="../reference/CircosPlot.html">CircosPlot</a></span>(<span class="no">test</span>,<span class="kw">sources.include</span> <span class="kw">=</span> <span class="st">'endothelial'</span>,<span class="kw">lab.cex</span> <span class="kw">=</span> <span class="fl">0.6</span>)</pre></body></html></div>
<p><img src="01%20Connectome%20Workflow_files/figure-html/unnamed-chunk-16-1.png" width="960" style="display: block; margin: auto;"></p>
</div>
<div id="large-scale-visualizations" class="section level2">
<h2 class="hasAnchor">
<a href="#large-scale-visualizations" class="anchor"></a>Large-scale visualizations</h2>
<p>Although it can be difficult to intepret, <code>Connectome</code> can also be used to provide large-scale overviews of whole-tissue signaling networks in scRNAseq data:</p>
<div class="sourceCode" id="cb17"><html><body><pre class="r"><span class="fu"><a href="../reference/CircosPlot.html">CircosPlot</a></span>(<span class="no">panc8.con2</span>,<span class="kw">min.z</span><span class="kw">=</span><span class="fl">1</span>,<span class="kw">lab.cex</span> <span class="kw">=</span> <span class="fl">0.4</span>,<span class="kw">gap.degree</span> <span class="kw">=</span> <span class="fl">0.1</span>)</pre></body></html></div>
<p><img src="01%20Connectome%20Workflow_files/figure-html/unnamed-chunk-17-1.png" width="1536" style="display: block; margin: auto;"></p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Micha Sam Brickman Raredon.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
