<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Differential Connectomics • Connectome</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Differential Connectomics">
<meta property="og:description" content="Connectome">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">Connectome</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.2.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/01%20Connectome%20Workflow.html">Connectome Basic Workflow</a>
    </li>
    <li>
      <a href="../articles/02%20Differential%20Connectomics.html">Differential Connectomics</a>
    </li>
    <li>
      <a href="../articles/03%20Centrality%20Plots.html">Centrality Plots</a>
    </li>
    <li>
      <a href="../articles/04%20Comparative%20and%20Longitudinal%20Connectomics.html">Comparative and Longitudinal Connectomics</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Differential Connectomics</h1>
                        <h4 class="author">Micha Sam Brickman Raredon</h4>
            
            <h4 class="date">2020-05-05</h4>
      
      
      <div class="hidden name"><code>02 Differential Connectomics.Rmd</code></div>

    </div>

    
    
<p><code>Connectome</code> can be used to identify and visualize major perturbed cell-cell communication pathways in A:B comparisons of complex tissue systems.</p>
<div id="load-dependencies" class="section level2">
<h2 class="hasAnchor">
<a href="#load-dependencies" class="anchor"></a>Load dependencies</h2>
<div class="sourceCode" id="cb1"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">Seurat</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">SeuratData</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">Connectome</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">ggplot2</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">cowplot</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">ComplexHeatmap</span>)</pre></body></html></div>
</div>
<div id="load-data" class="section level2">
<h2 class="hasAnchor">
<a href="#load-data" class="anchor"></a>Load Data</h2>
<p>To demonstrate differential connectomics using <code>Connectome</code>, we will use the interferon-stimulated vs. control PBMC data distributed by SeuratData:</p>
<div class="sourceCode" id="cb2"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/pkg/SeuratData/man/InstallData.html">InstallData</a></span>(<span class="st">'ifnb'</span>)
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="st">'ifnb'</span>)
<span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span>(<span class="fu"><a href="https://rdrr.io/pkg/Seurat/man/Idents.html">Idents</a></span>(<span class="no">ifnb</span>))
<span class="co">#&gt; </span>
<span class="co">#&gt; IMMUNE_CTRL IMMUNE_STIM </span>
<span class="co">#&gt;        6548        7451</span>
<span class="fu"><a href="https://rdrr.io/pkg/Seurat/man/Idents.html">Idents</a></span>(<span class="no">ifnb</span>) <span class="kw">&lt;-</span> <span class="no">ifnb</span><span class="kw">[[</span><span class="st">'seurat_annotations'</span>]]
<span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span>(<span class="fu"><a href="https://rdrr.io/pkg/Seurat/man/Idents.html">Idents</a></span>(<span class="no">ifnb</span>))
<span class="co">#&gt; </span>
<span class="co">#&gt;    CD14 Mono  CD4 Naive T CD4 Memory T    CD16 Mono            B </span>
<span class="co">#&gt;         4362         2504         1762         1044          978 </span>
<span class="co">#&gt;        CD8 T  T activated           NK           DC  B Activated </span>
<span class="co">#&gt;          814          633          619          472          388 </span>
<span class="co">#&gt;           Mk          pDC        Eryth </span>
<span class="co">#&gt;          236          132           55</span></pre></body></html></div>
</div>
<div id="scale-and-make-connectomes" class="section level2">
<h2 class="hasAnchor">
<a href="#scale-and-make-connectomes" class="anchor"></a>Scale and make connectomes</h2>
<p>Connectomic networks must first be calculated for each tissue system separately. To do so:</p>
<div class="sourceCode" id="cb3"><html><body><pre class="r"><span class="co"># First identify ligands and receptors which have mapped in the dataset:</span>
<span class="no">connectome.genes</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sets.html">union</a></span>(<span class="kw pkg">Connectome</span><span class="kw ns">::</span><span class="no">ncomms8866_human</span>$<span class="no">Ligand.ApprovedSymbol</span>,<span class="kw pkg">Connectome</span><span class="kw ns">::</span><span class="no">ncomms8866_human</span>$<span class="no">Receptor.ApprovedSymbol</span>)
<span class="no">genes</span> <span class="kw">&lt;-</span> <span class="no">connectome.genes</span>[<span class="no">connectome.genes</span> <span class="kw">%in%</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="no">ifnb</span>)]
<span class="co"># Split the object by condition:</span>
<span class="no">ifnb.list</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Seurat/man/SplitObject.html">SplitObject</a></span>(<span class="no">ifnb</span>,<span class="kw">split.by</span> <span class="kw">=</span> <span class="st">'stim'</span>)
<span class="co"># Normalize, Scale, and create Connectome:</span>
<span class="no">ifnb.con.list</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span>()
<span class="kw">for</span> (<span class="no">i</span> <span class="kw">in</span> <span class="fl">1</span>:<span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span>(<span class="no">ifnb.list</span>)){
  <span class="no">ifnb.list</span><span class="kw">[[</span><span class="no">i</span>]] <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Seurat/man/NormalizeData.html">NormalizeData</a></span>(<span class="no">ifnb.list</span><span class="kw">[[</span><span class="no">i</span>]])
  <span class="no">ifnb.list</span><span class="kw">[[</span><span class="no">i</span>]] <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Seurat/man/ScaleData.html">ScaleData</a></span>(<span class="no">ifnb.list</span><span class="kw">[[</span><span class="no">i</span>]],<span class="kw">features</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="no">ifnb.list</span><span class="kw">[[</span><span class="no">i</span>]]))
  <span class="no">ifnb.con.list</span><span class="kw">[[</span><span class="no">i</span>]] <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/CreateConnectome.html">CreateConnectome</a></span>(<span class="no">ifnb.list</span><span class="kw">[[</span><span class="no">i</span>]],<span class="kw">species</span> <span class="kw">=</span> <span class="st">'human'</span>,<span class="kw">p.values</span> <span class="kw">=</span> <span class="no">F</span>)
}
<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span>(<span class="no">ifnb.con.list</span>) <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span>(<span class="no">ifnb.list</span>)</pre></body></html></div>
</div>
<div id="make-differential-connectome" class="section level2">
<h2 class="hasAnchor">
<a href="#make-differential-connectome" class="anchor"></a>Make differential connectome</h2>
<p>The two global signaling networks can then be directly compared. Here, we do so using the function <code>DifferentialConnectome</code>. For this functon to run, the two systems must be parcellated in exactly the same way (the same cell types must be present in both) and the rownames (genes) of the original DGE matrices must be identical.</p>
<div class="sourceCode" id="cb4"><html><body><pre class="r"><span class="no">diff</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/DifferentialConnectome.html">DifferentialConnectome</a></span>(<span class="no">ifnb.con.list</span><span class="kw">[[</span><span class="fl">1</span>]],<span class="no">ifnb.con.list</span><span class="kw">[[</span><span class="fl">2</span>]])</pre></body></html></div>
</div>
<div id="initial-scoring-plot" class="section level2">
<h2 class="hasAnchor">
<a href="#initial-scoring-plot" class="anchor"></a>Initial Scoring Plot</h2>
<p>Differential connectomics are challenging to visualize, because every differential edge takes data from 4 places: the ligand expression in both the control and test condition, and the receptor expression in the control and test condition. Further, a single ligand can hit multiple receptors and vice-versa. One way to visualize this data is with edge-aligned heatmaps. Here, we use the function <code>DifferentialScoringPlot</code> to create three aligned heatmaps: one showing the ligand log fold change, one showing the receptor log fold change, and one showing a ‘perturbation score’ which is the product of the absolute values of the data in the first two plots. This helps to identify those edges which both pass the desired expression thresholds in at least one of the systems, and are highly perturbed in the A:B comparison.</p>
<div class="sourceCode" id="cb5"><html><body><pre class="r"><span class="fu"><a href="../reference/DifferentialScoringPlot.html">DifferentialScoringPlot</a></span>(<span class="no">diff</span>,<span class="kw">min.score</span> <span class="kw">=</span> <span class="fl">2</span>,<span class="kw">min.pct</span> <span class="kw">=</span> <span class="fl">0.1</span>,<span class="kw">infinity.to.max</span> <span class="kw">=</span> <span class="no">T</span>)</pre></body></html></div>
<p><img src="02%20Differential%20Connectomics_files/figure-html/unnamed-chunk-5-1.png" width="4800" style="display: block; margin: auto;"></p>
</div>
<div id="differential-circos-plots" class="section level1">
<h1 class="hasAnchor">
<a href="#differential-circos-plots" class="anchor"></a>Differential Circos Plots</h1>
<p>Since the above plot is not particularly concise, we have built <code>DifferentialCircos</code>, which draws circos plots for a differential connectome. This plot type makes use of the perturbation score, which is <strong>always</strong> positive, by design. However, in conenctomic data, there are 4 separate classes of perturbations, as both the ligand and the receptor can either be differentially increased or decreased. Here, we look at the above edgelist (which is shown in full in the DifferentialScoringPlot) and investigate each type of perturbation in turn:</p>
<div id="ligand-and-receptor-are-both-up" class="section level2">
<h2 class="hasAnchor">
<a href="#ligand-and-receptor-are-both-up" class="anchor"></a>Ligand and receptor are both <strong>UP</strong>:</h2>
<div class="sourceCode" id="cb6"><html><body><pre class="r"><span class="no">diff.up.up</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Seurat/man/subset.Seurat.html">subset</a></span>(<span class="no">diff</span>,<span class="no">ligand.norm.lfc</span> <span class="kw">&gt;</span> <span class="fl">0</span> <span class="kw">&amp;</span> <span class="no">recept.norm.lfc</span> <span class="kw">&gt;</span> <span class="fl">0</span> )
<span class="fu"><a href="../reference/CircosDiff.html">CircosDiff</a></span>(<span class="no">diff.up.up</span>,<span class="kw">min.score</span> <span class="kw">=</span> <span class="fl">2</span>,<span class="kw">min.pct</span> <span class="kw">=</span> <span class="fl">0.1</span>,<span class="kw">lab.cex</span> <span class="kw">=</span> <span class="fl">0.4</span>)</pre></body></html></div>
<p><img src="02%20Differential%20Connectomics_files/figure-html/unnamed-chunk-6-1.png" width="960" style="display: block; margin: auto;"></p>
</div>
<div id="ligand-is-up-and-receptor-is-down" class="section level2">
<h2 class="hasAnchor">
<a href="#ligand-is-up-and-receptor-is-down" class="anchor"></a>Ligand is <strong>UP</strong> and receptor is <strong>DOWN</strong>:</h2>
<div class="sourceCode" id="cb7"><html><body><pre class="r"><span class="no">diff.up.down</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Seurat/man/subset.Seurat.html">subset</a></span>(<span class="no">diff</span>,<span class="no">ligand.norm.lfc</span> <span class="kw">&gt;</span> <span class="fl">0</span> <span class="kw">&amp;</span> <span class="no">recept.norm.lfc</span> <span class="kw">&lt;</span> <span class="fl">0</span> )
<span class="fu"><a href="../reference/CircosDiff.html">CircosDiff</a></span>(<span class="no">diff.up.down</span>,<span class="kw">min.score</span> <span class="kw">=</span> <span class="fl">2</span>,<span class="kw">min.pct</span> <span class="kw">=</span> <span class="fl">0.1</span>,<span class="kw">lab.cex</span> <span class="kw">=</span> <span class="fl">0.4</span>)</pre></body></html></div>
<p><img src="02%20Differential%20Connectomics_files/figure-html/unnamed-chunk-7-1.png" width="960" style="display: block; margin: auto;"></p>
</div>
<div id="ligand-is-down-and-receptor-is-up" class="section level2">
<h2 class="hasAnchor">
<a href="#ligand-is-down-and-receptor-is-up" class="anchor"></a>Ligand is <strong>DOWN</strong> and receptor is <strong>UP</strong>:</h2>
<div class="sourceCode" id="cb8"><html><body><pre class="r"><span class="no">diff.down.up</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Seurat/man/subset.Seurat.html">subset</a></span>(<span class="no">diff</span>,<span class="no">ligand.norm.lfc</span> <span class="kw">&lt;</span> <span class="fl">0</span> <span class="kw">&amp;</span> <span class="no">recept.norm.lfc</span> <span class="kw">&gt;</span> <span class="fl">0</span> )
<span class="fu"><a href="../reference/CircosDiff.html">CircosDiff</a></span>(<span class="no">diff.down.up</span>,<span class="kw">min.score</span> <span class="kw">=</span> <span class="fl">2</span>,<span class="kw">min.pct</span> <span class="kw">=</span> <span class="fl">0.1</span>,<span class="kw">lab.cex</span> <span class="kw">=</span> <span class="fl">0.4</span>)</pre></body></html></div>
<p><img src="02%20Differential%20Connectomics_files/figure-html/unnamed-chunk-8-1.png" width="960" style="display: block; margin: auto;"></p>
</div>
<div id="ligand-and-receptor-are-both-down" class="section level2">
<h2 class="hasAnchor">
<a href="#ligand-and-receptor-are-both-down" class="anchor"></a>Ligand and receptor are both <strong>DOWN</strong>:</h2>
<div class="sourceCode" id="cb9"><html><body><pre class="r"><span class="no">diff.down.down</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Seurat/man/subset.Seurat.html">subset</a></span>(<span class="no">diff</span>,<span class="no">ligand.norm.lfc</span> <span class="kw">&lt;</span> <span class="fl">0</span> <span class="kw">&amp;</span> <span class="no">recept.norm.lfc</span> <span class="kw">&lt;</span> <span class="fl">0</span> )
<span class="fu"><a href="../reference/CircosDiff.html">CircosDiff</a></span>(<span class="no">diff.down.down</span>,<span class="kw">min.score</span> <span class="kw">=</span> <span class="fl">2</span>,<span class="kw">min.pct</span> <span class="kw">=</span> <span class="fl">0.1</span>,<span class="kw">lab.cex</span> <span class="kw">=</span> <span class="fl">0.4</span>)</pre></body></html></div>
<p><img src="02%20Differential%20Connectomics_files/figure-html/unnamed-chunk-9-1.png" width="960" style="display: block; margin: auto;"></p>
</div>
</div>
<div id="focusing-on-specific-cell-cell-interactions" class="section level1">
<h1 class="hasAnchor">
<a href="#focusing-on-specific-cell-cell-interactions" class="anchor"></a>Focusing on specific cell-cell interactions:</h1>
<p>We can also use <code>DifferentialCircos</code> to quantify edge perturbation within a specific subset or community of cells:</p>
<div class="sourceCode" id="cb10"><html><body><pre class="r"><span class="fu"><a href="../reference/CircosDiff.html">CircosDiff</a></span>(<span class="no">diff</span>,<span class="kw">min.score</span> <span class="kw">=</span> <span class="fl">2</span>,<span class="kw">min.pct</span> <span class="kw">=</span> <span class="fl">0.1</span>,<span class="kw">lab.cex</span> <span class="kw">=</span> <span class="fl">0.4</span>,
           <span class="kw">sources.include</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">'pDC'</span>,<span class="st">'CD8 T'</span>,<span class="st">'B'</span>),<span class="kw">targets.include</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">'CD16 Mono'</span>,<span class="st">'CD14 Mono'</span>))</pre></body></html></div>
<p><img src="02%20Differential%20Connectomics_files/figure-html/unnamed-chunk-10-1.png" width="960" style="display: block; margin: auto;"></p>
</div>
<div id="quantifing-pertubations-in-a-cellular-niche" class="section level1">
<h1 class="hasAnchor">
<a href="#quantifing-pertubations-in-a-cellular-niche" class="anchor"></a>Quantifing pertubations in a cellular ‘niche’…</h1>
<p>By subsetting to only those edges falling on a specific cell type, we can quantify the effect of a systemic perturbation (in this specific case, interferon treatment) on a given cellular niche:</p>
<div class="sourceCode" id="cb11"><html><body><pre class="r"><span class="fu"><a href="../reference/CircosDiff.html">CircosDiff</a></span>(<span class="no">diff</span>,<span class="kw">min.score</span> <span class="kw">=</span> <span class="fl">2</span>,<span class="kw">min.pct</span> <span class="kw">=</span> <span class="fl">0.1</span>,<span class="kw">lab.cex</span> <span class="kw">=</span> <span class="fl">0.4</span>,
           <span class="kw">targets.include</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">'CD14 Mono'</span>))</pre></body></html></div>
<p><img src="02%20Differential%20Connectomics_files/figure-html/unnamed-chunk-11-1.png" width="960" style="display: block; margin: auto;"></p>
</div>
<div id="or-on-a-single-receptor-on-a-specific-cell-type" class="section level1">
<h1 class="hasAnchor">
<a href="#or-on-a-single-receptor-on-a-specific-cell-type" class="anchor"></a>..or on a single receptor on a specific cell type</h1>
<p>Since the above plot is still somehwat complex, we can further focus on the network landing on a single receptor, on a single cell type:</p>
<div class="sourceCode" id="cb12"><html><body><pre class="r"><span class="fu"><a href="../reference/CircosDiff.html">CircosDiff</a></span>(<span class="no">diff</span>,<span class="kw">min.score</span> <span class="kw">=</span> <span class="fl">2</span>,<span class="kw">min.pct</span> <span class="kw">=</span> <span class="fl">0.1</span>,<span class="kw">lab.cex</span> <span class="kw">=</span> <span class="fl">0.6</span>,
           <span class="kw">targets.include</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">'CD14 Mono'</span>),<span class="kw">features</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">'CCR5'</span>))</pre></body></html></div>
<p><img src="02%20Differential%20Connectomics_files/figure-html/unnamed-chunk-12-1.png" width="960" style="display: block; margin: auto;"></p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Micha Sam Brickman Raredon.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
